api-client/src/main/java/com/xing/api/resources/UserProfilesResource.java

```diff
@@ -34,6 +34,12 @@
  * @author daniel.hartwich
  */
 public class UserProfilesResource extends Resource {
+    public static final Resource.Factory FACTORY = new Resource.Factory() {
+        @Override public Resource create(Class<? extends Resource> cls, XingApi api) {
+            return cls == UserProfilesResource.class ? new UserProfilesResource(api) : null;
```

Comment:
This code is repeated many times, wouldn't make sense to let Resource implementations just return a new instance and do the class check in the base Resource class?